src/Moq/ExpressionExtensions.cs

```diff
@@ -262,6 +278,13 @@ void Split(Expression e, out Expression r /* remainder */, out InvocationShape p
 						return;
 					}
 
+					case ExpressionType.Extension when (e is AwaitExpression awaitExpression):
+					{
+						Split(awaitExpression.Operand, out r, out p);
+						p.AwaitableHandler = awaitExpression.AwaitableHandler;
+						return;
```

Comment:
An `AwaitExpression` appears to get thrown away here. This might cause the "await" to disappear from the resulting setup's `Expression`.